data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetEth1VotesStats.java

```diff
@@ -0,0 +1,213 @@
+import tech.pegasys.teku.spec.datastructures.blocks.Eth1Data;
+import tech.pegasys.teku.validator.coordinator.Eth1DataProvider;
+import tech.pegasys.teku.validator.coordinator.Eth1DataProvider.VotingPeriodInfo;
+
+public class GetEth1VotesStats extends MigratingEndpointAdapter {
+
+  public static final String OAPI_ROUTE = "/teku/v1/beacon/state/:state_id/eth1stats";
+  public static final String ROUTE = routeWithBracedParameters(OAPI_ROUTE);
+  public static final String NOT_FOUND_MESSAGE = "Corresponding state not found";
+
+  private static final SerializableTypeDefinition<Pair<Eth1Data, UInt64>> ETH1DATA_WITH_VOTES_TYPE =
+      SerializableTypeDefinition.<Pair<Eth1Data, UInt64>>object()
+          .name("Eth1DataWithVotes")
+          .withField("eth1_data", Eth1Data.SSZ_SCHEMA.getJsonTypeDefinition(), Pair::getLeft)
+          .withField("votes", CoreTypes.UINT64_TYPE, Pair::getRight)
+          .build();
+
+  private static final SerializableTypeDefinition<Eth1DataVotesStats> ETH1VOTES_STATS_TYPE =
+      SerializableTypeDefinition.<Eth1DataVotesStats>object()
+          .name("Eth1VotesStats")
+          .withField(
+              "eth1_data",
+              Eth1Data.SSZ_SCHEMA.getJsonTypeDefinition(),
+              Eth1DataVotesStats::getEth1Data)
+          .withField(
+              "eth1_data_votes",
+              listOf(ETH1DATA_WITH_VOTES_TYPE),
+              Eth1DataVotesStats::getVotesBreakdown)
+          .withField(
+              "win_votes_required", CoreTypes.UINT64_TYPE, Eth1DataVotesStats::getWinVotesRequired)
+          .withField(
+              "voting_period_slots",
+              CoreTypes.UINT64_TYPE,
+              Eth1DataVotesStats::getVotingPeriodSlots)
+          .withField(
+              "voting_period_slots_left",
+              CoreTypes.UINT64_TYPE,
+              Eth1DataVotesStats::getVotingPeriodSlotsLeft)
+          .build();
+
+  private static final SerializableTypeDefinition<Eth1DataVotesStats>
+      ETH1VOTES_STATS_RESPONSE_TYPE =
+          SerializableTypeDefinition.<Eth1DataVotesStats>object()
+              .name("GetEth1VotesStatsResponse")
+              .withField("data", ETH1VOTES_STATS_TYPE, Function.identity())
+              .build();
+
+  private final ChainDataProvider chainDataProvider;
+  private final Eth1DataProvider eth1DataProvider;
+
+  public GetEth1VotesStats(
+      final DataProvider dataProvider, final Eth1DataProvider eth1DataProvider) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getEth1VotesStats")
+            .summary("Get Eth1DataVotes stats")
+            .description(
+                "Returns information about the current state of voting for Eth1Data from the specified state.")
+            .pathParam(PARAMETER_STATE_ID)
+            .tags(TAG_TEKU)
+            .response(SC_OK, "Request successful", ETH1VOTES_STATS_RESPONSE_TYPE)
+            .response(SC_NOT_FOUND, NOT_FOUND_MESSAGE)
```

Comment:
This declares that the 404 response will have no content so it fails when the 404 response actually has content.  Currently with a checkpoint sync node:
```
curl http://localhost:5051/teku/v1/beacon/state/12/eth1stats
```

will return a 500 server error.

Can just use the helper function to get this right:
```suggestion
            .withNotFoundResponse()
```

The unit test wasn't flagging the problem because the stub request didn't actually check the response content matched what was declared in the metadata.  I've fixed that so the test should start failing now when you pull in the latest changes.  It is important to actually check new functionality by actually running it and doing some exploratory testing though as it shows up any potential gaps in test coverage like this.

Reply:
My fault I haven't checked this and just copied. I see now that some code is old and could be really not tested to blindly trust it. Will keep this in my mind